emcc.py

```diff
@@ -1591,14 +1591,23 @@ def phase_setup(options, state, newargs):
     settings.DISABLE_EXCEPTION_CATCHING = 0
 
   if settings.WASM_EXCEPTIONS:
-    if 'DISABLE_EXCEPTION_CATCHING' in user_settings:
-      exit_with_error('DISABLE_EXCEPTION_CATCHING is not compatible with -fwasm-exceptions')
-    if 'DISABLE_EXCEPTION_THROWING' in user_settings:
-      exit_with_error('DISABLE_EXCEPTION_THROWING is not compatible with -fwasm-exceptions')
+    # DISABLE_EXCEPTION_CATCHING is 1 by default. So if it is 0, it means the
+    # user specifies it explicitly
+    if not settings.DISABLE_EXCEPTION_CATCHING:
+      exit_with_error('DISABLE_EXCEPTION_CATCHING=0 is not compatible with -fwasm-exceptions')
+    # DISABLE_EXCEPTION_THROWING is 0 by default, but in Wasm EH, it defaults to
+    # 1. If the user explicitly gives DISABLE_EXCEPTION_THROWING=0, error out.
+    if 'DISABLE_EXCEPTION_THROWING' in user_settings and user_settings['DISABLE_EXCEPTION_THROWING'] == '0':
+      exit_with_error('DISABLE_EXCEPTION_THROWING=0 is not compatible with -fwasm-exceptions')
+    # -fwasm-exceptions takes care of enabling them, so users aren't supposed to
+    # pass them explicitly
+    if 'DISABLE_EXCEPTION_CATCHING' in user_settings or 'DISABLE_EXCEPTION_THROWING' in user_settings:
+      diagnostics.warning('emcc', 'You no longer need to pass DISABLE_EXCEPTION_CATCHING or DISABLE_EXCEPTION_THROWING when using Wasm exceptions')
+    default_setting('DISABLE_EXCEPTION_CATCHING', 1)
+    default_setting('DISABLE_EXCEPTION_THROWING', 1)
```

Comment:
I think its clearer to just unconditionally set these like in the existing code.

Reply:
Done. But when do you use the unconditional setting and when this `default_setting`? I saw many places using `default_setting` so I did it too, but not sure what the difference is in the intentions.

Reply:
`default_setting` should only be used when you want to let the user override the setting.  If you want to unconditionally set a setting you don't use that.

`default_setting` basically just does "set the setting unless the user already set it, in which case leave it alone".

In this case we know we don't want to use the control these settings, so I think its obvious the reader if we just unconditionally set them.